subsys/bluetooth/audio/has.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <device.h>
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/gatt.h>
+#include <bluetooth/audio/audio.h>
+#include <bluetooth/audio/capabilities.h>
+
+#include "has_internal.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_HAS)
+#define LOG_MODULE_NAME bt_has
+#include "common/log.h"
+
+static struct bt_has has;
+
+static ssize_t read_features(struct bt_conn *conn, const struct bt_gatt_attr *attr, void *buf,
+			     uint16_t len, uint16_t offset)
+{
+	BT_DBG("conn %p attr %p offset %d", conn, attr, offset);
+
+	if (offset > sizeof(has.features)) {
+		return BT_GATT_ERR(BT_ATT_ERR_INVALID_OFFSET);
+	}
+
+	return bt_gatt_attr_read(conn, attr, buf, len, offset, &has.features,
+				 sizeof(has.features));
+}
+
+/* Hearing Access Service GATT Attributes */
+BT_GATT_SERVICE_DEFINE(has_svc,
+	BT_GATT_PRIMARY_SERVICE(BT_UUID_HAS),
+	BT_GATT_CHARACTERISTIC(BT_UUID_HAS_HEARING_AID_FEATURES, BT_GATT_CHRC_READ,
+			       BT_GATT_PERM_READ_ENCRYPT, read_features, NULL, NULL),
+);
+
+static int has_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	/* Initialize the supported features characteristic value */
+	has.features = CONFIG_BT_HAS_HEARING_AID_TYPE;
```

Comment:
This will need to be masked (sooner or later) - the type is only two bits of the feature characteristic.  Do that already now?

Reply:
> This will need to be masked (sooner or later) - the type is only two bits of the feature characteristic. Do that already now?

If `CONFIG_BT_HAS_HEARING_AID_TYPE` only contains valid values, why do we need a mask? 

If we need to add anything, it should be an __ASSERT (better to fail explicitly than to implicitly mask invalid values imo).

Reply:
I agree with @Thalley

Reply:
The Hearing Aid Features characteristic (one octet) consists of five fields (plus two spare bits), of which the Hearing Aid Type is one.  When the remaining fields are implemented some way of setting one field without overwriting the others will be needed.
I am OK to leave this as is for now, though.